api-registration/server/index.js

```diff
@@ -42,15 +42,19 @@ app.post('/api/auth/sign-up', (req, res, next) => {
       const sql = `
         insert into "users" ("username", "hashedPassword")
              values ($1, $2)
-             returning *
+             returning "userId", "username", "createdAt"
       `;
       const params = [username, hashedPassword];
       db.query(sql, params)
         .then(result => {
           res.status(201).json(result.rows[0]);
+          if (!username || !password) {
+            throw new ClientError(400, 'username and password are required fields');
+          }
```

Comment:
This shouldn't be necessary. We've already checked this when the request first came in. It's too late anyway because we've already inserted the user into the db.

Reply:
true, sorry, brain not working